packages/components/src/internal/components/search/actions.ts

```diff
@@ -328,3 +339,50 @@ export function getSampleTypesFromFindByIdQuery(
             });
     });
 }
+
+export function getSampleFinderTabRowCounts(queryModels: {
+    [key: string]: QueryModel;
+}): Promise<{ [key: string]: number }> {
+    const modelIds = Object.keys(queryModels);
+    const sampleTypeGridIds = {};
+    let allSamplesModel = null;
+    const tabCounts = {};
+    modelIds.forEach(modelId => {
+        const model = queryModels[modelId];
+        if (model.schemaQuery.schemaName === SCHEMAS.SAMPLE_SETS.SCHEMA) {
+            sampleTypeGridIds[model.schemaQuery.queryName.toLowerCase()] = modelId;
+        } else if (model.schemaQuery.schemaName === EXP_TABLES.MATERIALS.schemaName) {
+            allSamplesModel = model;
+        }
+        tabCounts[modelId] = 0;
+    });
+
+    return new Promise((resolve, reject) => {
+        Query.executeSql({
+            containerFilter: getContainerFilter(),
+            schemaName: SCHEMAS.EXP_TABLES.SCHEMA,
+            sql: getSampleFinderTabRowCountSql(allSamplesModel),
+            success: result => {
+                const typeCounts = {};
+                result.rows?.forEach(row => {
+                    const type = caseInsensitive(row, 'SampleTypeName');
+                    typeCounts[type] = caseInsensitive(row, 'RowCount');
+                });
+
+                let totalCount = 0;
```

Comment:
The count for All samples is done by summing up all counts on each tab, so it's not adding to the query time. In an unlikely scenario, a user (or automated test) could click on a different tab before All tab starts loading, in which case the calculated row count is still helpful. I'm inclined to leave this in since there is no negative influence with including it.

Reply:
Issue filed https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=47503

Fixed in this branch